cpp/include/cudf/table/experimental/row_operators.cuh

```diff
@@ -427,6 +433,162 @@ class self_comparator {
   std::shared_ptr<preprocessed_table> d_t;
 };
 
+template <typename Nullate>
+class two_table_device_row_comparator_adapter {
+  friend class two_table_comparator;
+
+ public:
+  /**
+   * @brief Checks whether the row at `lhs_index` in the `lhs` table compares
+   * lexicographically less than the row at `rhs_index` in the `rhs` table.
+   *
+   * @param lhs_index The index of row in the `lhs` table to examine
+   * @param rhs_index The index of the row in the `rhs` table to examine
+   * @return `true` if row from the `lhs` table compares less than row in the `rhs` table
+   */
+  __device__ bool operator()(lhs_index_type const lhs_index,
+                             rhs_index_type const rhs_index) const noexcept
```

Comment:
I think we'll need overloads for `(lhs_index_type, lhs_index_type)` and `(rhs_index_type, rhs_index_type)` as well.

Reply:
Possible — but I haven’t found an algorithm that needs those yet. For example, merge requires sorted inputs, so the lhs/lhs and rhs/rhs overloads are never needed.

Reply:
Also, overloads for `(lhs_index_type, lhs_index_type)` and `(rhs_index_type, rhs_index_type)` would require this class to own two additional comparators (self comparators for left and right tables).

Reply:
Hey guys! I stumbled on to this: using `self_comparator` with strong index types! So we are going to need `(lhs_index_type, lhs_index_type)` and `(rhs_index_type, rhs_index_type)` overloads for `self_comparator`.